### ISSUE TITLE
efficientnetb0 has less trainable parametrs then resnet18 and resnet34 but consumes more memory

### ISSUE THREAD

**Issue by Diyago** [qubvel/segmentation_models#128]
Simple Unet model efficientnetb0 has:
_ Total params: 10,115,501
 Trainable params: 10,071,501
 Non-trainable params: 44,000_

resnet18:
_Total params: 14,340,570
Trainable params: 14,330,644
Non-trainable params: 9,926_

However first model fail to run with input image 512+256 and batch size 4, while second one works like a charm. Is that okay?

**Comment by qubvel** [comment]
Hi @Diyago 
Did you mean train or test mode?

**Comment by Diyago** [comment]
train mode

**Comment by qubvel** [comment]
In train mode we have to store intermediate computations for all layers (to compute backprop). In efficientnet `swish` activation is used which is much more memory inefficient than `relu`.
Less parameters not always mean less memory consumption.
